polkadot/primitives/src/v5/executor_params.rs

```diff
@@ -130,6 +130,16 @@ impl ExecutorParams {
 		}
 		None
 	}
+
+	/// Returns pre-checking memory limit, if any
+	pub fn prechecking_max_memory(&self) -> Option<u64> {
+		for param in &self.0 {
+			if let ExecutorParam::PrecheckingMaxMemory(limit) = param {
+				return Some(*limit)
+			}
```

Comment:
That is not enforced, you may still have `[PrecheckingMaxMemory(1024), PrecheckingMaxMemory(2048)]`, and the first one will be in effect. But I believe in this case, what is questionable is not the program logic but the motivation of someone who proposed it for the referenda and the governance action accepting this obviously erroneous change.

Reply:
Well you know, people make mistakes. :P If we could defend against such erroneous changes somehow it would be nice... We could at least check and warn (somewhere, maybe not here), to raise awareness of the issue, so that it can be investigated and fixed by governance.

Reply:
Wrong thread, but relates to both anyway :sweat_smile:

Reply:
> I'll make it `.try_into().expect()` for now, but I've raised #1704 as well. Thanks for being insistent on security-related questions 🙂

No problem! You think a panic in this situation is okay? I was just thinking to log a warning on invalid values and reset them to a default.